### DISCUSSION TITLE
How to take first n elements in array?

### DISCUSSION THREAD

**Discussion by balki** [mikefarah/yq#1294] [Q&A]
Example:

```
❯ cat ex.yaml                                                   
- foo
- bar
- baz
- blah
```

Trying to take first 2 elements

```
❯ yq '.[0:2]' ex.yaml
Error: cannot index array with '' (strconv.ParseInt: parsing "": invalid syntax)

❯ yq '.[0 2]' ex.yaml
Error: Bad expression, please check expression syntax

❯ yq '.[0..2]' ex.yaml
Error: Bad expression, please check expression syntax

```
This works though but not easy for large n

```
❯ yq '.[0,1,2]' ex.yaml 
foo
bar
baz

```

This works but wondering if there is a concise way i am missing.

❯ yq '.[] | select(key < 2)' ex.yaml 
foo
bar

**Comment by mikefarah** [comment]
Hi Balki, adding support for this jq's slice syntax `.[0:2]` is on the TODO list (https://github.com/mikefarah/yq/issues/44)

Till then, there isn't a great way to do it - though your trick of `.[] | select(key < 2)` is quite cool. You may want to wrap it in `[]` to put it back into an array `[.[] | select(key < 2)]`.

**Reply by balki** [reply]
Thank you! Is it ok If I try to fix and send a PR?

**Reply by mikefarah** [reply]
For anything else I'd say go for it, I'm a little hesitant with this as it's fairly deep in `yq`'s hairy core logic - the underlying `.[]` operator is quite complex, it handles things like `.[.name, .cat]?`, it'll need to be updated to handle these slices without breaking other random bits...

**Reply by mikefarah** [reply]
If you're looking for a feature to work on - I'd love help with https://github.com/mikefarah/yq/discussions/1298 - basically yq needs a new merge flag (like [these](https://mikefarah.gitbook.io/yq/operators/multiply-merge#merge-flags)) called `t` that will make it clobber existing tags (basically disabling the check in UpdateAttributesFrom ``if strings.HasPrefix(n.Node.Tag, "!!") || n.Node.Tag == ""` 

It's still reasonably hairy because multiple/merge is the most complex operator...

**Reply by balki** [reply]
Thank you for the help and suggesting an issue to work on. At the moment, I can't contribute with PR - a bit too complex and not much free time. May be will do in future. Thank you for the amazing tool.
